jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js

```diff
@@ -485,52 +478,51 @@ var module = module;
       deploying a bundle via the deployer.
 
       @method getBundleYAML
-      @param {String} id Bundle id in apiv4 format.
-      @param {Function} successCallback The success callback.
-      @param {Function} failureCallback The failure callback.
+      @param id {String} Bundle id in apiv4 format.
+      @params callback {Function} A callback to handle errors or accept the data
+          from the request. Must accept an error message or null as its first
+          parameter and the response data as its second.
     */
-    getBundleYAML: function(id, successCallback, failureCallback) {
+    getBundleYAML: function(id, callback) {
       this.getEntity(
-          id, this._getBundleYAMLResponse.bind(
-              this, successCallback, failureCallback), failureCallback);
+          id, this._getBundleYAMLResponse.bind(this, callback));
     },
 
     /**
       getEntity success response handler which grabs the deployerFileUrl from
-      the recieved bundle details and requests the YAML.
+      the received bundle details and requests the YAML.
 
       @method _getBundleYAMLResponse
-      @param {Function} successCallback The success callback.
-      @param {Function} failureCallback The failure callback.
-      @param {Array} bundle An array containing the requested bundle model.
-      @return {Object} The asynchronous request instance.
+      @params callback {Function} A callback to handle errors or accept the data
+          from the request. Must accept an error message or null as its first
+          parameter and the response data as its second.
+      @param error {String} An error message or null if no error.
+      @param bundle {Array} An array containing the requested bundle model.
     */
-    _getBundleYAMLResponse: function(successCallback, failureCallback, bundle) {
-      return this._makeRequest(
-          bundle[0].get('deployerFileUrl'),
-          function(resp) {
-            successCallback(resp.currentTarget.responseText);
-          },
-          failureCallback);
+    _getBundleYAMLResponse: function(callback, error, bundle) {
+      return _makeRequest(
+          this.bakery, bundle[0].get('deployerFileUrl'), 'GET',
+          null, callback, false);
     },
 
     /**
       Gets the list of available versions of the supplied charm id.
 
       @method getAvailableVersions
-      @param {String} charmId The charm id to fetch all of the versions for.
-      @param {Function} successCallback The success callback.
-      @param {Function} failureCallback The failure callback.
-      @return {Object} The asynchronous request instance.
+      @param charmId {String} The charm id to fetch all of the versions for.
```

Comment:
Deliberately; I've seen a lot of js modules put the name of the param before the type which makes it a bit clearer that the name isn't part of the description.